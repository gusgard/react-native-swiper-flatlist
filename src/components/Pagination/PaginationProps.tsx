import { ViewStyle } from 'react-native';

export type PaginationProps = {
  /**
   * Scroll to the index
   *
   */
  scrollToIndex: (item: { index: number }) => void;
  /**
   * Size of pagination
   *
   */
  size: number;
  /**
   * Selected pagination index
   * Defaults to '0'
   *
   */
  paginationIndex?: number;
  /**
   * Pagination color
   * Defaults to 'white'
   *
   */
  paginationActiveColor?: string;
  /**
   * Pagination color
   * Defaults to 'gray'
   *
   */
  paginationDefaultColor?: string;
  /**
   * Style object for the container
   * Defaults to '{}'
   *
   */
  paginationStyle?: ViewStyle;
  /**
   * Style object for the item (dot)
   * Defaults to '{}'
   *
   */
  paginationStyleItem?: ViewStyle;
  /**
   * Style object for the active item (dot)
   * Defaults to '{}'
   *
   */
  paginationStyleItemActive?: ViewStyle;
  /**
   * Style object for the inactive item (dot)
   * Defaults to '{}'
   *
   */
  paginationStyleItemInactive?: ViewStyle;
  /**
   * Executed when the user presses the pagination index, similar properties onChangeIndex
   *
   */
  onPaginationSelectedIndex?: () => void;
  /**
   * TestID for automation testing
   *
   */
  e2eID?: string;
};
