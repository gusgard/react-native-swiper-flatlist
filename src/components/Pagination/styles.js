import { StyleSheet } from 'react-native';

import { vertical, horizontal, width } from '../../themes';

export default StyleSheet.create({
  paginationContainer: {
    position: 'absolute',
    flexDirection: 'row',
    marginVertical: vertical.xxSmall,
    justifyContent: 'center',
    bottom: 0,
    left: width * 0.25,
    right: width * 0.25,
  },
  pagination: {
    width: horizontal.small,
    height: horizontal.small,
    borderRadius: 25,
    marginHorizontal: horizontal.xSmall,
  },
});
