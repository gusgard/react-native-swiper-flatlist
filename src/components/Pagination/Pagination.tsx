import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { colors, vertical, horizontal } from '../../themes';
import { PaginationProps } from './PaginationProps';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    flexDirection: 'row',
    marginVertical: vertical.xxSmall,
    justifyContent: 'center',
    alignSelf: 'center',
    bottom: 0,
    height: horizontal.small,
  },
  pagination: {
    width: horizontal.small,
    height: horizontal.small,
    borderRadius: 25,
    marginHorizontal: horizontal.xSmall,
  },
});

export const Pagination: React.FC<PaginationProps> = ({
  size,
  paginationIndex = 0,
  scrollToIndex,
  paginationDefaultColor = colors.gray,
  paginationActiveColor = colors.white,
  paginationStyle = {},
  paginationStyleItem = {},
  paginationStyleItemActive = {},
  paginationStyleItemInactive = {},
  onPaginationSelectedIndex,
  e2eID = '',
}) => {
  return (
    <View style={[styles.container, paginationStyle]}>
      {Array.from({ length: size }).map((_, index) => (
        <TouchableOpacity
          testID={`${e2eID}_pagination_${index}`}
          style={[
            styles.pagination,
            paginationStyleItem,
            paginationIndex === index
              ? { backgroundColor: paginationActiveColor }
              : { backgroundColor: paginationDefaultColor },
            paginationIndex === index ? paginationStyleItemActive : paginationStyleItemInactive,
          ]}
          key={index}
          onPress={() => {
            scrollToIndex({ index });
            onPaginationSelectedIndex?.();
          }}
        />
      ))}
    </View>
  );
};
