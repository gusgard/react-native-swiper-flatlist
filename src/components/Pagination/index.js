import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, View } from 'react-native';

import { colors } from '../../themes';
import styles from './styles';

export default class Pagination extends PureComponent {
  static propTypes = {
    scrollToIndex: PropTypes.func.isRequired,
    data: PropTypes.array,
    paginationIndex: PropTypes.number,
    paginationActiveColor: PropTypes.string,
    paginationDefaultColor: PropTypes.string
  };

  static defaultProps = {
    data: [],
    paginationIndex: 0,
    paginationActiveColor: colors.white,
    paginationDefaultColor: colors.gray
  };

  render() {
    const {
      data,
      paginationIndex,
      scrollToIndex,
      paginationDefaultColor,
      paginationActiveColor
    } = this.props;
    return (
      <View style={styles.paginationContainer}>
        {data.map((_, index) => (
          <TouchableOpacity
            style={[
              styles.pagination,
              paginationIndex === index
                ? { backgroundColor: paginationActiveColor }
                : { backgroundColor: paginationDefaultColor }
            ]}
            key={index}
            onPress={() => scrollToIndex(index)}
          />
        ))}
      </View>
    );
  }
}
