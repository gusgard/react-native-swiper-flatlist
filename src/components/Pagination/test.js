import React from 'react';
import { render } from 'react-native-testing-library';

import { Pagination } from './Pagination';

describe('pagination', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Pagination scrollToIndex={() => undefined} size={5} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders all props', () => {
    const { toJSON } = render(
      <Pagination
        scrollToIndex={() => undefined}
        paginationIndex={1}
        paginationActiveColor="black"
        paginationDefaultColor="white"
        size={5}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
