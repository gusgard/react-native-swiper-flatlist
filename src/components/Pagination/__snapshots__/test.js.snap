// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination renders all props 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 16.675,
        "position": "absolute",
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
</View>
`;

exports[`pagination renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 16.675,
        "position": "absolute",
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 25,
        "height": 28.125,
        "marginHorizontal": 18.75,
        "opacity": 1,
        "width": 28.125,
      }
    }
  />
</View>
`;
